blocks/lists.ts

```diff
@@ -138,53 +132,56 @@ blocks['lists_create_with'] = {
   /**
    * Create XML to represent list inputs.
    * Backwards compatible serialization implementation.
-   * @return {!Element} XML storage element.
-   * @this {Block}
    */
-  mutationToDom: function() {
+  mutationToDom: function(this: CreateWithBlock): Element {
     const container = xmlUtils.createElement('mutation');
     container.setAttribute('items', this.itemCount_);
     return container;
   },
   /**
    * Parse XML to restore the list inputs.
    * Backwards compatible serialization implementation.
-   * @param {!Element} xmlElement XML storage element.
-   * @this {Block}
+   *
+   * @param xmlElement XML storage element.
    */
-  domToMutation: function(xmlElement) {
-    this.itemCount_ = parseInt(xmlElement.getAttribute('items'), 10);
+  domToMutation: function(this: CreateWithBlock, xmlElement: Element) {
+    const items = xmlElement.getAttribute('items');
+    if (items === null) throw new TypeError('element did not have items');
+    this.itemCount_ = parseInt(items, 10);
     this.updateShape_();
   },
   /**
    * Returns the state of this block as a JSON serializable object.
-   * @return {{itemCount: number}} The state of this block, ie the item count.
+   *
+   * @return The state of this block, ie the item count.
    */
-  saveExtraState: function() {
+  saveExtraState: function(): {itemCount: number} {
```

Comment:
Why doesn't this one need a `this:` parameter?

Reply:
Only because the body does not actually reference `this`.  It could still have a `this:` though.  Do you think it is better to do so, for consistency?

Reply:
I thought it does reference `this.itemCount_`, which is why I was confused.

Reply:
Oh, so it does.  I will correct (there are some other issues too.)